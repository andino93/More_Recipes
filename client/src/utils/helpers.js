import jwtDecode from 'jwt-decode';
import history from '../utils/history';

const decodeToken = token => jwtDecode(token);

const logout = () => {
  window.sessionStorage.removeItem('token');
  history.push('/');
};

export {
  decodeToken,
  logout
};
