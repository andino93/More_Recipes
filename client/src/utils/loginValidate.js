import Validator from 'validator';
import isEmpty from 'lodash/isEmpty';

const loginValidate = (data) => {
  const errors = {};

  if (Validator.isEmpty(data.email)) {
    errors.email = 'Email is required';
  }

  if (Validator.isEmpty(data.password)) {
    errors.password = 'Password is required';
  }

  return {
    errors,
    isValid: isEmpty(errors)
  };
};

export default loginValidate;
