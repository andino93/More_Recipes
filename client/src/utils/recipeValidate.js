import Validator from 'validator';
import isEmpty from 'lodash/isEmpty';

const recipeValidate = (data) => {
  const name = data.name.trim(),
    description = data.description.trim(),
    ingredient = data.ingredient.trim();
  const errors = {};

  if (!Validator.isEmpty(name)) {
    const hasNumber = name.split('').filter(character => Validator.toInt(character));
    if (hasNumber.length !== 0) {
      errors.name = 'Recipe name must contain alphabetic characters only';
    }
  } else { errors.name = 'Recipe name is required'; }

  if (!Validator.isEmpty(description)) {
    if (!Validator.isLength(description, { min: 4, max: undefined })) {
      errors.description = 'Recipe description must be at least 4 characters';
    }
  } else { errors.description = 'Recipe description is required'; }

  if (!Validator.isEmpty(ingredient)) {
    if (!Validator.isLength(ingredient, { min: 3, max: undefined })) {
      errors.ingredient = 'Recipe ingredient must be at least 3 characters';
    }
  } else { errors.ingredient = 'Recipe ingredient is required'; }

  return {
    errors,
    isValid: isEmpty(errors)
  };
};

export default recipeValidate;