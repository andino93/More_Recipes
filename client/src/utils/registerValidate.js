import Validator from 'validator';
import isEmpty from 'lodash/isEmpty';

const registerValidate = (data) => {
  const errors = {};
  
  if (Validator.isEmpty(data.firstName)) {
    errors.firstName = 'First name is required';
  }

  if (Validator.isEmpty(data.lastName)) {
    errors.lastName = 'Last name is required';
  }

  if (Validator.isEmpty(data.username)) {
    errors.username = 'Username is required';
  }

  if (Validator.isEmpty(data.email)) {
    errors.email = 'Email is required';
  }

  if (Validator.isEmpty(data.password)) {
    errors.password = 'Password is required';
  }

  if (Validator.isEmpty(data.confirmPassword)) {
    errors.confirmPassword = 'Confirm password is required';
  }

  if (!Validator.equals(data.password, data.confirmPassword)) {
    errors.confirmPassword = 'Passwords must match';
  }

  return {
    errors,
    isValid: isEmpty(errors)
  };
};

export default registerValidate;
