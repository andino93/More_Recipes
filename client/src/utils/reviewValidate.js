import Validator from 'validator';
import isEmpty from 'lodash/isEmpty';

const reviewValidate = (data) => {
  const content = data.content.trim();
  const errors = {};

  if (!Validator.isEmpty(content)) {
    if (!Validator.isLength(content, { min: 4, max: undefined })) {
      errors.content = 'Review  must be at least 4 characters';
    }
  } else { errors.content = 'Review details is required'; }

  return {
    errors,
    isValid: isEmpty(errors)
  };
};

export default reviewValidate;