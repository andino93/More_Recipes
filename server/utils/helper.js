const errorHandler = (error, res) => {
  switch (error.code) {
    case 404:
      return res.status(404).json({ message: error.message });
    case 500:
      return res.status(400).json({ message: error.message });
    default:
      return res.status(500).json(error);
  }
};

const successHandler = (code, body, res) => {
  switch (code) {
    case 201:
      return res.status(201).json({ body });
    default:
      return res.status(200).json({ body });
  }
};

const generatePaginationMeta = (dbResult, limit, offset) => {
  const paginationMeta = {
    pageCount: Math.ceil(dbResult.count / limit),
    totalCount: dbResult.count,
    outputCount: dbResult.rows.length,
    pageSize: limit,
    currentPage: Math.floor(offset / limit) + 1
  };
  return paginationMeta;
};

export { errorHandler, successHandler, generatePaginationMeta };
