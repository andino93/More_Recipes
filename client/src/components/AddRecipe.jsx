import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'react-proptypes';
import classnames from 'classnames';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Spinner from "react-md-spinner";
import {
  processRecipeActions,
  clearToast
} from '../actions/recipe/recipeActions';
import recipeValidate from '../utils/recipeValidate';
import FooterComponent from './partials/Footer.jsx';
import Header from './partials/Headers/Header.jsx';
import { recipeResponseType } from '../utils/helpers';


/**
 *
 * @export
 * @class AddRecipe
 * @extends {React.Component}
 */
export class AddRecipe extends Component {
  /**
   * Creates an instance of AddRecipeComponent.
   * @param {any} props
   * @memberof AddRecipeComponent
   */
  constructor(props) {
    super(props);
    this.state = {
      recipeDetails: {
        name: '',
        description: '',
        ingredient: '',
      },
      errors: {},
      imageFile: '',
      imageSrc: '/assets/images/no-image.jpg'
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleImageChange = this.handleImageChange.bind(this);
  }
  /**
   * @return {nextProps} nextProps
   * @param {nextProps} nextProps
   */
  componentWillReceiveProps(nextProps) {
    const { message, responseType } = nextProps.recipeState;
    if (responseType === recipeResponseType.ADD_RECIPE_SUCCESS && message !== '') {
      toastr.clear();
      toastr.success(message);
      this.props.clearToast();
    } else if (message !== '') {
      toastr.clear();
      toastr.error(message);
      this.props.clearToast();
    }
  }
  /**
   * Handle submit
   *
   * @return {event} event
   * @param {event} event
   */
  handleSubmit(event) {
    event.preventDefault();
    if (this.checkValidity()) {
      this.setState({ errors: {} });
      this.props.processRecipeActions(this.state.recipeDetails, this.state.imageFile);
      this.setState({
        recipeDetails: {
          name: '',
          description: '',
          ingredient: ''
        },
        imageSrc: '/assets/images/no-image.jpg'
      });
    }
  }
  /**
   * Handle Image Upload
   *
   * @return {event} event
   * @param {event} event
   */
  handleImageChange(event) {
    if (event.target.files && event.target.files[0]) {
      this.setState({ imageFile: event.target.files[0] });
      const reader = new FileReader();
      reader.onload = (e) => {
        this.setState({ imageSrc: e.target.result });
      };
      reader.readAsDataURL(event.target.files[0]);
    } else {
      this.setState({
        imageSrc: '/assets/images/no-image.jpg',
        imageFile: ''
      });
    }
  }
  /**
   * Check Validity
   *
   * @return {object} event
   */
  checkValidity() {
    const { errors, isValid } = recipeValidate(this.state.recipeDetails);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }

  /**
   * Handle change
   *
   * @return {event} event
   * @param {event} event
   */
  handleChange(event) {
    const recipeDetails = this.state.recipeDetails;
    recipeDetails[event.target.name] = event.target.value;
    this.setState({ recipeDetails });
  }
  /**
   * 
   * 
   * @returns {XML} XML/JSX
   * @memberof AddRecipeComponent
   */
  render() {
    const { recipeDetails, imageSrc, errors } = this.state;
    const { recipeState } = this.props;
    return (
      <div>
        <Header/>
        <div className="wrapper">
          <ol
            className="breadcrumb container mt-50 mb-10 col-md-9 mx-auto bg-white shadow-lite">
            <li className="breadcrumb-item">
              <Link to="/">Home</Link>
            </li>
            <li className="breadcrumb-item">
              <Link to="#">Recipes</Link>
            </li>
            <li className="breadcrumb-item active">Add new</li>
          </ol>
          <div className="container mb-20 col-md-9 mx-auto recipe-details-container">
            <div className="row">
              <div
                className="col-12 col-sm-6 col-md-4 col-lg-4 p-10 shadow-lite bg-white"
                style={{
                  height: `${400}px`,
                  maxHeight: `${400}px`
                }}>
                <div
                  className="light-well center-content"
                  style={{
                    height: `${100}%`
                  }}>
                  <div className="text-center file-upload full-height full-width center-content">
                    <input
                      type="file"
                      className="recipe-upload text-hide"
                      onChange={this.handleImageChange}
                      accept="image/*" />
                    <img src={imageSrc} alt="sample" height="400" width="100%"/>
                  </div>
                </div>
              </div>
              <div
                className="simplebox blade col-12 col-sm-6 col-md-8 col-lg-8 p-10"
                style={{
                  boxShadow: 'none'
                }}>
                <h5>Recipe Details</h5>
                <form className="mt-15 recipe-form" onSubmit={this.handleSubmit}>
                  { recipeState.fails ?
                    <div id="error" className="alert alert-danger alert-dismissible">
                      <button type="button"
                        className="close"
                        data-dismiss="alert"
                        aria-label="Close">
                        <span aria-hidden="true">x</span>
                      </button>
                      { recipeState.fails }
                    </div> : null
                  }
                  <div className="form-group">
                    <input type="text"
                      className={
                        classnames('form-control',
                          { 'is-invalid': errors.name ? !!errors.name : false })
                      }
                      name="name"
                      aria-describedby="name"
                      onChange={this.handleChange}
                      value={recipeDetails.name}
                      placeholder="Recipe Name"
                    />
                    { errors.name ?
                      <span className="invalid-feedback">{ errors.name }</span> : null
                    }
                  </div>
                  <h6>Procedure</h6>
                  <div className="form-group">
                    <textarea
                      className={
                        classnames('form-control',
                          {
                            'is-invalid': errors.description ? !!errors.description : false
                          })
                      }
                      rows="4"
                      name="description"
                      onChange={this.handleChange}
                      value={recipeDetails.description}
                      placeholder="Recipe description here" />
                    { errors.description ?
                      <span className="invalid-feedback">{ errors.description }</span> : null
                    }
                  </div>
                  <h6>Ingredients</h6>
                  <div className="form-group">
                    <textarea className={classnames('form-control', {
                      'is-invalid': errors.ingredient ? !!errors.ingredient : false
                    })}
                    rows="4"
                    placeholder="Ingredient1, ingredient2..."
                    name="ingredient"
                    value={recipeDetails.ingredient}
                    onChange={this.handleChange}
                    />
                    <span className="small text-info">Please enter comma separated values</span>
                    { errors.ingredient ?
                      <span className="invalid-feedback">{ errors.ingredient }</span> : null
                    }
                  </div>
                  <button type="submit"
                    className="btn btn-outline-success"
                    disabled={recipeState.isCreating}
                  >
                    {recipeState.isCreating ? 
                      <span>Adding Recipe... <Spinner size={20} />
                      </span> : 'Add Recipe' }
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
        <FooterComponent />
      </div>
    );
  }
}

AddRecipe.propTypes = {
  recipeState: PropTypes.object.isRequired,
  processRecipeActions: PropTypes.func.isRequired,
  clearToast: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  recipeState: state.recipeReducer
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({ processRecipeActions, clearToast }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(AddRecipe);
