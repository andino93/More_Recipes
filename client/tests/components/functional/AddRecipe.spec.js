import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import sinon from 'sinon';
import mockStore from '../../__mocks__/mockStore';
import mountWrap from '../../__mocks__/utils';
import localStorageMock from '../../__mocks__/localStorageMock';
import AddRecipe from '../../../src/components/AddRecipe.jsx';
import { recipeResponseType } from '../../../src/utils/helpers';
import mockData from "../../__mocks__/mockData";

configure({ adapter: new Adapter() });

// noinspection JSAnnotator
window.localStorage = localStorageMock;
window.localStorage.token = mockData.token;

describe('<AddRecipe>', () => {
  const store = mockStore({ recipeReducer: { recipeState: {} } });
  const history = {
    push: () => {}
  };
  const connectedAddRecipe = mountWrap(AddRecipe, store, {
    history,
  });
  it('should render the AddRecipe Component', (done) => {
    expect(toJson(connectedAddRecipe)).toMatchSnapshot();
    done();
  });
  it('should check for componentWillReceiveProps', () => {
    const pushSpy = sinon.spy(history, 'push');
    connectedAddRecipe.find('AddRecipe').instance().componentWillReceiveProps({
      recipeState: {
        message: 'Recipe successfully added',
        responseType: recipeResponseType.ADD_RECIPE_SUCCESS,
        recipeId: 4,
      },
    });
    expect(pushSpy.called).toEqual(true);
  });

  it('should check for componentWillReceiveProps', () => {
    connectedAddRecipe.find('AddRecipe').instance().componentWillReceiveProps({
      recipeState: {
        message: 'Recipe successfully added',
        responseType: null,
        recipeId: 4,
      },
    });
  });

  it('should simulate handleSubmit', (done) => {
    connectedAddRecipe.find('form').at(0).simulate('submit', {
      preventDefault: () => {}
    });
    done();
  });

  it('should simulate handleChange', (done) => {
    const { recipe } = mockData.updateRecipeResponse;
    connectedAddRecipe.find('AddRecipe').instance().setState({
      recipeDetails: { ...recipe }
    });
    connectedAddRecipe.find('form').find('input').at(0).simulate('change', {
      preventDefault: () => {},
      target: {
        name: 'name',
        value: 'Updated recipe'
      }
    });
    expect(connectedAddRecipe.find('AddRecipe').instance().state.recipeDetails.name).toEqual('Updated recipe');
    done();
  });

  it('should return true after checkValidity function call', (done) => {
    const { recipe } = mockData.updateRecipeResponse;
    connectedAddRecipe.find('AddRecipe').instance().setState({
      recipeDetails: { ...recipe }
    });
    expect(connectedAddRecipe.find('AddRecipe').instance().checkValidity()).toEqual(true);
    connectedAddRecipe.find('AddRecipe').instance().setState({
      recipeDetails: { ...recipe, name: '' }
    });
    expect(connectedAddRecipe.find('AddRecipe').instance().checkValidity()).toEqual(false);
    done();
  });
});
