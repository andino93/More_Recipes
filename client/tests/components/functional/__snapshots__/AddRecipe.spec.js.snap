// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddRecipe> should render the AddRecipe Component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(AddRecipe)
        history={
          Object {
            "push": [Function],
          }
        }
      >
        <AddRecipe
          clearToast={[Function]}
          history={
            Object {
              "push": [Function],
            }
          }
          processRecipeActions={[Function]}
          recipeState={
            Object {
              "recipeState": Object {},
            }
          }
        >
          <div>
            <Connect(Header)>
              <Header
                logoutAction={[Function]}
              >
                <div>
                  <nav
                    className="navbar navbar-expand-sm navbar-light bg-navbar fixed-top"
                  >
                    <Link
                      className="navbar-brand"
                      replace={false}
                      to="/"
                    >
                      <a
                        className="navbar-brand"
                        href="/"
                        onClick={[Function]}
                      >
                        More Recipes
                      </a>
                    </Link>
                    <button
                      aria-controls="navbarSupportedContent"
                      aria-expanded="false"
                      aria-label="Toggle navigation"
                      className="navbar-toggler"
                      data-target="#navbarSupportedContent"
                      data-toggle="collapse"
                      type="button"
                    >
                      <span
                        className="navbar-toggler-icon"
                      />
                    </button>
                    <div
                      className="collapse navbar-collapse"
                      id="navbarSupportedContent"
                    >
                      <ul
                        className="navbar-nav ml-auto"
                      >
                        <li
                          className="nav-item dropdown"
                        >
                          <Link
                            aria-expanded="false"
                            aria-haspopup="true"
                            className="nav-link dropdown-toggle"
                            data-toggle="dropdown"
                            id="navbarDropdown"
                            replace={false}
                            role="button"
                            to="#"
                          >
                            <a
                              aria-expanded="false"
                              aria-haspopup="true"
                              className="nav-link dropdown-toggle"
                              data-toggle="dropdown"
                              href="#"
                              id="navbarDropdown"
                              onClick={[Function]}
                              role="button"
                            >
                              My Account
                            </a>
                          </Link>
                          <div
                            aria-labelledby="navbarDropdown"
                            className="dropdown-menu dropdown-menu-right mr-20"
                          >
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/profile"
                            >
                              <a
                                className="dropdown-item"
                                href="/profile"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-user-circle"
                                />
                                 My Profile
                              </a>
                            </Link>
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/user-recipes"
                            >
                              <a
                                className="dropdown-item"
                                href="/user-recipes"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-list"
                                />
                                 My Recipes
                              </a>
                            </Link>
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/my-favorite"
                            >
                              <a
                                className="dropdown-item"
                                href="/my-favorite"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-star"
                                />
                                 My Favorites
                              </a>
                            </Link>
                            <Link
                              className="dropdown-item"
                              replace={false}
                              to="/addRecipe"
                            >
                              <a
                                className="dropdown-item"
                                href="/addRecipe"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-plus"
                                />
                                 Add Recipes
                              </a>
                            </Link>
                            <div
                              className="dropdown-divider"
                            />
                            <Link
                              className="dropdown-item"
                              onClick={[Function]}
                              replace={false}
                              to="#"
                            >
                              <a
                                className="dropdown-item"
                                href="#"
                                onClick={[Function]}
                              >
                                <i
                                  className="fa fa-sign-out"
                                />
                                Log out
                              </a>
                            </Link>
                          </div>
                        </li>
                      </ul>
                    </div>
                  </nav>
                </div>
              </Header>
            </Connect(Header)>
            <div
              className="wrapper"
            >
              <ol
                className="breadcrumb container mt-50 mb-10 col-md-9 mx-auto bg-white shadow-lite"
              >
                <li
                  className="breadcrumb-item"
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      Home
                    </a>
                  </Link>
                </li>
                <li
                  className="breadcrumb-item"
                >
                  <Link
                    replace={false}
                    to="#"
                  >
                    <a
                      href="#"
                      onClick={[Function]}
                    >
                      Recipes
                    </a>
                  </Link>
                </li>
                <li
                  className="breadcrumb-item active"
                >
                  Add new
                </li>
              </ol>
              <div
                className="container mb-20 col-md-9 mx-auto recipe-details-container"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-12 col-sm-6 col-md-4 col-lg-4 p-10 shadow-lite bg-white"
                    style={
                      Object {
                        "height": "400px",
                        "maxHeight": "400px",
                      }
                    }
                  >
                    <div
                      className="light-well center-content"
                      style={
                        Object {
                          "height": "100%",
                        }
                      }
                    >
                      <div
                        className="text-center file-upload full-height full-width center-content"
                      >
                        <input
                          accept="image/*"
                          className="recipe-upload text-hide"
                          onChange={[Function]}
                          type="file"
                        />
                        <img
                          alt="sample"
                          height="400"
                          width="100%"
                        />
                      </div>
                    </div>
                    <br />
                    <span
                      className="text-info"
                    >
                      Please click to upload an image  
                      <i
                        className="fa fa-upload"
                      />
                    </span>
                  </div>
                  <div
                    className="simplebox blade col-12 col-sm-6 col-md-8 col-lg-8 p-10"
                    style={
                      Object {
                        "boxShadow": "none",
                      }
                    }
                  >
                    <h5>
                      Recipe Details
                    </h5>
                    <form
                      className="mt-15 recipe-form"
                      onSubmit={[Function]}
                    >
                      <div
                        className="form-group"
                      >
                        <input
                          aria-describedby="name"
                          className="form-control"
                          name="name"
                          onChange={[Function]}
                          placeholder="Recipe Name"
                          type="text"
                          value=""
                        />
                      </div>
                      <h6>
                        Procedure
                      </h6>
                      <div
                        className="form-group"
                      >
                        <textarea
                          className="form-control"
                          name="description"
                          onChange={[Function]}
                          placeholder="Recipe description here"
                          rows="4"
                          value=""
                        />
                      </div>
                      <h6>
                        Ingredients
                      </h6>
                      <div
                        className="form-group"
                      >
                        <textarea
                          className="form-control"
                          name="ingredient"
                          onChange={[Function]}
                          placeholder="Ingredient1, ingredient2..."
                          rows="4"
                          value=""
                        />
                        <span
                          className="small text-info"
                        >
                          Please enter comma separated values
                        </span>
                      </div>
                      <button
                        className="btn btn-outline-success"
                        type="submit"
                      >
                        Add Recipe
                      </button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
            <FooterComponent>
              <footer
                className="p-10 well well-sm m-0 shadow-lite"
              >
                <div
                  className="text-center text-muted"
                >
                  <p
                    className="bold"
                  >
                    © 2017 More Recipes
                  </p>
                </div>
              </footer>
            </FooterComponent>
          </div>
        </AddRecipe>
      </Connect(AddRecipe)>
    </Router>
  </BrowserRouter>
</Provider>
`;
